File: weather-app-android/app/src/main/java/org/pitkanen/weather_app_android/api/WeatherApi.kt
package org.pitkanen.weather_app_android.api

import org.pitkanen.weather_app_android.model.Model
import retrofit2.Call
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.GET
import retrofit2.http.Query

interface WeatherApi {
    @GET("/data/2.5/weather")
    fun getWeatherData(@Query("APPID") APPID: String,
                     @Query("lat") lat: String,
                     @Query("lon") lon: String,
                     @Query("units") units: String): Call<Model.WeatherData>

    @GET("/data/2.5//forecast")
    fun getForecastData(@Query("APPID") APPID: String,
                       @Query("lat") lat: String,
                       @Query("lon") lon: String,
                       @Query("units") units: String): Call<Model.ForecastData>

    companion object Factory {
        fun create(): WeatherApi {
            val retrofit = Retrofit.Builder()
                    .baseUrl("https://api.openweathermap.org/")
                    .addConverterFactory(GsonConverterFactory.create())
                    .build()

            return retrofit.create(WeatherApi::class.java)
        }
    }
}

File: weather-app-flutter/lib/api/WeatherApi.dart
import 'package:http/http.dart' as http;

class WeatherApi {

  static const END_POINT = 'https://api.openweathermap.org/data/2.5';
  static const APP_ID = '0613fdf35a1e7d5548e0d2b3b11296cb';
  static const IMAGE_END_POINT = 'https://openweathermap.org/img/w/';
  static const FILE_TYPE = '.png';
  static const CELSIUS = '°C';

  static getWeatherData(final lat, final lon) {
    return http.get('$END_POINT/weather?APPID=$APP_ID&lat=$lat&lon=$lon&units=metric');
  }

  static getForecastData(final lat, final lon) {
    return http.get('$END_POINT/forecast?APPID=$APP_ID&lat=$lat&lon=$lon&units=metric');
  }

  static getWeatherIconUrl(final icon) {
    return '$IMAGE_END_POINT$icon$FILE_TYPE';
  }
}